README.md

```diff
@@ -0,0 +1,264 @@
+![image](https://initialcommit.com/img/initialcommit/baby-git-release.png)
+
+# 1. Git Basics:
+
+## What is Version Control System?
+
+- It is a software that helps software developers to work together and maintain a complete history of their work.
+
+## Types of VCS:
+
+1. CVCS : Centralized version control system.
+
+- it uses a central server to store all files. 
+- The major drawback of CVCS is if the disk of the central server gets corrupted then you will lose the entire history of the project
+
+![image](https://user-images.githubusercontent.com/125242584/219282906-5b90643d-6354-47b3-8349-29250edb548f.png)
+
+2. DVCS : Distributed/Decentralized version control system .
+
+- DVCS clients check out the latest snapshot of the directory and fully mirror the repository.
+- If the server goes down, then the repository from any client can be copied back to the server to restore it. Every checkout is a full backup of the repository.
+
+![image](https://user-images.githubusercontent.com/125242584/219284486-f2403e19-d4cc-4e27-959c-26a29317955b.png)
+
+
+## What is Git?
+
+- Git is Distributed version control system.
+- Git is open source project.
+
+## Workflow of Git:
+
+![image](https://user-images.githubusercontent.com/125242584/219288681-8bbcc71c-6a9c-4393-89a3-57dc1a267331.png)
+
+## Installation of Git Client:
+
+- For GNU/Linux.
+
+  [ubuntu ~]$ sudo apt-get install git-core </br>
+  [sudo] password for ubuntu:
+
+  [ubuntu ~]$ git --version</br>
+  git version 1.8.1.2
+
+## Git Environment Setup:
+
+1. Setting UserName:
+- `git config --global user.name "gunjan2012"`
+
+2. Setting email id:
+- `git config --global user.email "gunjan.d@simformsolutions.com"`
+
+3. Listing Git settings:
+- `git config --list`
+
+## Git General workflow:
+
+![image](https://www.tutorialspoint.com/git/images/life_cycle.png)
+
+# 2. Setting up a repository:
+
+## What is a Git repository?
+
+- A Git repository is a virtual storage of your project. It allows you to save versions of your code, which you can access when needed.
+
+## Initializing a new Git repo:
+`git init` : Intializing the new repository. </br>
+
+## Cloning an existing Git repo:
+`git clone` : Copy the existing repository.
+
+## Cloning specific branch:
+`git clone --branch` 
+
+## Deleting Git Repository:
+`rm -rf .git` : Deleting the .git folder from our local.
+
+## Rename file:
+`git mv old_name new_name`
+
+## Ignoring files:
```

Comment:
Please update the description of `gitignore`. We generally don't push `passwords`, `private-keys` to git repo, because they are important. also we don't push `node_module`. Please do more research and update the description accordingly.